### PR TITLE
batch submit deposit

### PR DESCRIPTION
**Pull request by jackzwc**
One batch per 16 deposits

### CODE REVIEW

File: internal/bls/handle.go [hunk 1]
```diff
@@ -296,7 +299,7 @@ func (s *Signer) handleSigStartNewDeposit(ctx context.Context, e types.MsgSignDe
 
 		// feedback SigFinish, deposit should module subscribe it to save UTXO or mark confirm
 		s.state.EventBus.Publish(state.SigFinish, e)
-		log.Infof("Proposer submit MsgNewDeposit to consensus ok, request id: %s", e.RequestId)
+		log.Info("Proposer submit MsgNewDeposit to consensus success")
```
Comment (aiden069) [inline internal/bls/handle.go:302]: Should add some data info like request id, deposit length...

File: internal/db/models.go [hunk 2]
```diff
@@ -206,6 +206,13 @@ type Deposit struct {
 	UpdatedAt   time.Time `gorm:"not null" json:"updated_at"`
 }
 
+type P2WSHInfo struct {
```
Comment (aiden069) [inline internal/db/models.go:209]: Remove this struct, use redeem info as a deposit field

File: internal/rpc/verifydeposit.go [hunk 3]
```diff
@@ -25,12 +24,13 @@ func (s *UtxoServer) VerifyDeposit(tx wire.MsgTx, evmAddress string) (isTrue boo
...
+	version := uint32(100)
+	pubKey, err := s.state.GetPubKey()
 	if err != nil {
-		return false, 100, err
+		return false, 100, "", err
 	}
 
-	version := uint32(100)
 	for _, out := range tx.TxOut {
 		pkScript := hex.EncodeToString(out.PkScript)
```
Comment (aiden069) [inline internal/rpc/verifydeposit.go:35]: Please use IsUtxoGoatDepositV1 directly if evmAddress is empy, else use IsUtxoGoatDepositV0
Many codes are not necerssary.

File: internal/state/deposit.go [hunk 4]
```diff
@@ -201,3 +201,21 @@ func (s *State) QueryBlockByTxHash(txHash string) (block *db.BtcBlockData, err e
 
 	return &btcBlockData, nil
 }
+
+// AddP2WSHInfo save p2wsh(v0) info in db
+func (s *State) AddP2WSHInfo(depositAddr string, evmAddr string, txId string) error {
```
Comment (aiden069) [inline internal/state/deposit.go:206]: Use deposit , add type for V0 and V1, add redeem script

File: internal/state/l2_info.go [hunk 5]
```diff
@@ -303,3 +304,28 @@ func (s *State) savePubKey(pubKey *db.DepositPubKey) error {
 	}
 	return nil
 }
+
+func (s *State) GetPubKey() ([]byte, error) {
```
Comment (aiden069) [inline internal/state/l2_info.go:308]: Deposit key is scanned from Event to layer2 state, try to use it.

File: internal/wallet/deposit.go [hunk 6]
```diff
@@ -110,39 +110,97 @@ func (w *WalletServer) confirmingDeposit(ctx context.Context, tx DepositTransact
...
 
-	isProposer := w.signer.IsProposer()
-	if isProposer {
-		l2Info := w.state.GetL2Info()
-		depositKey, err := hex.DecodeString(l2Info.DepositKey)
-		if err != nil {
-			log.Errorf("DecodeString DepositKey err: %v", err)
-			return
-		}
+	w.depositBatchCh <- DepositInfo{
```
Comment (aiden069) [inline internal/wallet/deposit.go:113]: I think it is better to fetch deposit from DB in a duration.

File: internal/db/models.go (hunk 2 above)
Comment (jackzwc) [reply internal/db/models.go]: done

**Pull request closed without merging**